Support Trino bitwise shift functions in Presto

djsstarburst opened the pull request:
To promote convergence between Trino and Presto, this commit
adds to Presto the bitwise functions present in Trino but missing
in Presto.

Test plan - Appropriate tests have been added to class TestBitwiseFunctions as part of this commit.

== RELEASE NOTES ==

General Changes
* The 12 integral shift functions defined by Trino's BitwiseFunctions class have been added to Presto's BitwiseFunctions class.  These shift functions represent a complete set of shift operations on tinyints, smallints, integers and longs.

aweisberg commented:
@Lewuathe appears to be the original author of this code. Please add them as a co-author. [See our attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).

djsstarburst commented:
> @Lewuathe appears to be the original author of this code. Please add them as a co-author. [See our attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).

Done.

aweisberg reviewed (approved):
LGTM. Checks were originally passing, but I imagine it has fallen far enough behind at this point that the FB integration check fails.

Thank you!

djsstarburst commented:
> LGTM. Checks were originally passing, but I imagine it has fallen far enough behind at this point that the FB integration check fails.

Thanks for pointing that out - - I just rebased on tip master and force-pushed, so that should fix it.

mbasmanova reviewed (changes requested):
@djsstarburst Would you add documentation for these functions?

djsstarburst commented:
> @djsstarburst Would you add documentation for these functions?

@mbasmanova, I added documentation and examples for the new shift functions in bitwise.rst.  While I was at it, I fixed the broken formatting in the examples of the previously existing shift functions.

mbasmanova reviewed (approved):
Thanks.

aweisberg commented:
@mbasmanova I restarted it and it looks like it is passing this time.

mbasmanova commented:
@djsstarburst Thank you, David, for the contribution. Thanks @aweisberg for restarting a failing job.
